main.cpp

```diff
@@ -423,6 +399,73 @@ IntType& IntType::divide(int rhs)
     return *this;
 }
 
+IntType& IntType::pow(int rhs)
+{
+    return powInternal(rhs);
+}
+
+IntType& IntType::pow(const FloatType& rhs)
+{
+    return powInternal(static_cast<int>(rhs));
+}
+
+IntType& IntType::pow(const DoubleType& rhs)
+{
+    return powInternal(static_cast<int>(rhs));
+}
+
+IntType& IntType::pow(const IntType& rhs)
+{
+    return powInternal(static_cast<int>(rhs));
+}
+
+IntType& IntType::powInternal(const int rhs)
+{
+    *value = static_cast<int>(std::pow(*value, rhs));
+    return *this;
+}
+
+struct Point
+{
+    Point(float x_, float y_) : x(x_), y(y_) { }
+    Point(double x_, double y_) : x(float(x_)), y(float(y_)) { }
+    Point(int x_, int y_) : x(float(x_)), y(float(y_)) { }
+
+    Point(FloatType& x_, FloatType& y_) : Point(float(x_), float(y_)) { }
+    Point(DoubleType& x_, DoubleType& y_) : Point(double(x_), double(y_)) { }
+    Point(IntType& x_, IntType& y_) : Point(int(x_), int(y_)) { }
```

Comment:
Don't use `C-Style` casting.
use `static_cast<>` and convert the arguments to `float`
remove the `Point(primitive, primitive)` constructors that don't take `float`
- [ ] revise